weblate/screenshots/models.py

```diff
@@ -76,3 +90,51 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    repository = component.repository
+    last_remote_revision = repository.last_remote_revision
+    changed_files = repository.get_changed_files(compare_to=previous_head)
+    screenshots = Screenshot.objects.filter(
+        translation__component=component, repository_filename__in=changed_files
+    )
+
+    # Update existing screenshots
+    for screenshot in screenshots:
+        filename = screenshot.repository_filename
+        changed_files.remove(filename)
+
+        file_data = repository.get_file(filename, last_remote_revision, raw=True)
+        image_file = File(file_data, name=filename)
+
+        try:
+            validate_bitmap(image_file)
+        except ValidationError:
+            report_error(cause="Failed to validate image from repository")
+            continue
+
+        screenshot.image = image_file
+        screenshot.save(update_fields=["image"])
```

Comment:
Will this move the file to the `screenshots` location, or just update the location filename to point to the VCS storage? In the latter case, this will break serving screenshots to the browser.